app/auth/authenticate_user.rb

```diff
@@ -0,0 +1,24 @@
+class AuthenticateUser
+  def initialize(email, password)
+    @email = email
+    @password = password
+  end
+
+  # Service entry point
+  def call
+    JsonWebToken.encode(user_id: user.id) if user
+  end
+
+  private
+
+  attr_reader :email, :password
+
+  # verify user credentials
+  def user
+    user = User.find_by(email: email)
+    return user if user&.authenticate(password)
+
+    # raise Authentication error if credentials are invalid
+    raise(ExceptionHandler::AuthenticationError, Message.invalid_credentials)
+  end
+end
```

Comment:
- [ ] Endpoint of the login or authentication doesn't work as expected. Please fix it.

![api4](https://user-images.githubusercontent.com/55635977/149813707-6d5b8e65-486a-46fd-8cc1-c169c9fbd9af.png)